src/Microsoft.TemplateSearch.TemplateDiscovery/Nuget/NugetPackProvider.cs

```diff
@@ -38,82 +41,64 @@ public NugetPackProvider(string packageTempBasePath, int pageSize, bool runOnlyO
             }
         }
 
-        public IEnumerable<IInstalledPackInfo> CandidatePacks
+        public async IAsyncEnumerable<IInstalledPackInfo> GetCandidatePacksAsync()
         {
-            get
-            {
-                int skip = 0;
-                bool done = false;
-                int packCount = 0;
+            int skip = 0;
+            bool done = false;
+            int packCount = 0;
 
-                do
+            do
+            {
+                string queryString = string.Format(SearchUrlFormat, skip, _pageSize, _includePreviewPacks);
+                Uri queryUri = new Uri(queryString);
+                using (HttpClient client = new HttpClient())
```

Comment:
addressing this would require implementing IDisposable in order to dispose HttpClient